### ISSUE TITLE
Unsupported locales problem. 

### ISSUE THREAD

**Issue by noqqe** [pimutils/khal#439]
Hi,

I get an error for  khal regarding the locales. I dont really understand the problem
~~~
$ khal

Traceback (most recent call last):
  File "/usr/local/bin/khal", line 11, in <module>
    sys.exit(main_khal())
  File "/usr/local/lib/python3.4/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.4/site-packages/click/core.py", line 1037, in invoke
    return Command.invoke(self, ctx)
  File "/usr/local/lib/python3.4/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.4/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/khal/cli.py", line 247, in cli
    ctx.invoke(cli.commands[command])
  File "/usr/local/lib/python3.4/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.4/site-packages/khal/cli.py", line 279, in calendar
    week=week,
  File "/usr/local/lib/python3.4/site-packages/khal/controllers.py", line 164, in calendar
    bold_for_light_color=bold_for_light_color)
  File "/usr/local/lib/python3.4/site-packages/khal/calendar_display.py", line 170, in vertical_month
    weekheaders = get_weekheader(firstweekday)
  File "/usr/local/lib/python3.4/site-packages/khal/calendar_display.py", line 47, in get_weekheader
    return _calendar.formatweekheader(2)
  File "/usr/local/lib/python3.4/calendar.py", line 301, in formatweekheader
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python3.4/calendar.py", line 301, in <genexpr>
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python3.4/calendar.py", line 515, in formatweekday
    with different_locale(self.locale):
  File "/usr/local/lib/python3.4/calendar.py", line 494, in __enter__
    _locale.setlocale(_locale.LC_TIME, self.locale)
  File "/usr/local/lib/python3.4/locale.py", line 592, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
~~~

My locale settings look as follows:

~~~
LANG=en_US.UTF-8
LC_COLLATE="C"
LC_CTYPE="en_US.UTF-8"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_MESSAGES="en_US.UTF-8"
LC_ALL=en_US.UTF-8
~~~

What do I have to reconfigure? Is en_US.UTF-8 not valid?

**Comment by geier** [comment]
Sorry for my late reply. This really looks all good in general. Can you try the following (and report back)?

```python
from locale import getlocale, setlocale, LC_ALL
import calendar

setlocale(LC_ALL, '')
mylocale = '.'.join(getlocale())
print(mylocale)
_calendar = calendar.LocaleTextCalendar(0, locale=mylocale)
print(_calendar.formatweekheader(2))
```

**Comment by geier** [comment]
@noqqe any updates?

**Comment by noqqe** [comment]
@geier sorry for the late reply.

```
en_US.UTF-8
Traceback (most recent call last):
  File "foo", line 8, in <module>
    print(_calendar.formatweekheader(2))
  File "/usr/local/lib/python2.7/calendar.py", line 301, in formatweekheader
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python2.7/calendar.py", line 301, in <genexpr>
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python2.7/calendar.py", line 516, in formatweekday
    with TimeEncoding(self.locale) as encoding:
  File "/usr/local/lib/python2.7/calendar.py", line 494, in __enter__
    _locale.setlocale(_locale.LC_TIME, self.locale)
  File "/usr/local/lib/python2.7/locale.py", line 579, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
```

Ant ideas?

**Comment by geier** [comment]
What does `locale` print in the shell?
 
And also please try executing the below python script:
```python
import calendar

_calendar = calendar.LocaleTextCalendar(0,  locale='en_US')
print(_calendar.formatweekheader(2))
```

**Comment by noqqe** [comment]
@geier mhpf. Sorry again for the bad response times.

```
11:52 noqqe@noc:~ $ python2 foo
Traceback (most recent call last):
  File "foo", line 4, in <module>
    print(_calendar.formatweekheader(2))
  File "/usr/local/lib/python2.7/calendar.py", line 301, in formatweekheader
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python2.7/calendar.py", line 301, in <genexpr>
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python2.7/calendar.py", line 516, in formatweekday
    with TimeEncoding(self.locale) as encoding:
  File "/usr/local/lib/python2.7/calendar.py", line 494, in __enter__
    _locale.setlocale(_locale.LC_TIME, self.locale)
  File "/usr/local/lib/python2.7/locale.py", line 579, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
```

and with pyhton3 

```
11:53 noqqe@noc:~ $ python3 foo
Traceback (most recent call last):
  File "foo", line 4, in <module>
    print(_calendar.formatweekheader(2))
  File "/usr/local/lib/python3.5/calendar.py", line 301, in formatweekheader
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python3.5/calendar.py", line 301, in <genexpr>
    return ' '.join(self.formatweekday(i, width) for i in self.iterweekdays())
  File "/usr/local/lib/python3.5/calendar.py", line 515, in formatweekday
    with different_locale(self.locale):
  File "/usr/local/lib/python3.5/calendar.py", line 494, in __enter__
    _locale.setlocale(_locale.LC_TIME, self.locale)
  File "/usr/local/lib/python3.5/locale.py", line 595, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
```

**Comment by noqqe** [comment]
Leads me to think its not your fault.. isnt it?

**Comment by geier** [comment]
@noqqe might be, but I'm not sure yet. As you have your LC_TIME locale explicitly set to C and we are forcing python to use the LANG settings, this might be a bad idea.

**Comment by geier** [comment]
if you have time for this, could you check out branch `fix_locale_problems` and see if your problems are gone?

**Comment by noqqe** [comment]
looks awesome!

![screenshot 2016-09-15 09 08 12](https://cloud.githubusercontent.com/assets/203902/18541213/f7aebaf4-7b23-11e6-92fb-295a8b3e9e24.png)

**Comment by geier** [comment]
thanks for helping resolving that issue!
should be fixed with  9f05c10

**Comment by noqqe** [comment]
Thanks alot for all your efforts!

**Comment by amracks** [comment]
I know you already merged this, but I had the same problem and 9f05c10 fixed it.

OS: OpenBSD 6.0
Locale:
LANG=en_US.UTF-8
LC_COLLATE="C"
LC_CTYPE=en_US.UTF-8
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_MESSAGES="en_US.UTF-8"
LC_ALL=

Thanks!

**Comment by buck-ross** [comment]
As of version 0.9.10 on Fedora Linux, I still had to run `export LC_ALL=C` to make it work. Not worth making a new issue IMO, but still worth noting.
